src/lake/Lake/Config/LeanLib.lean

```diff
@@ -87,6 +87,17 @@ Is true if either the package or the library have `precompileModules` set.
 @[inline] def nativeFacets (self : LeanLib) : Array (ModuleFacet (BuildJob FilePath)) :=
   self.config.nativeFacets
 
+/--
+The arguments to pass to `lean --server` when running the Lean language server.
+`serverOptions` is the the accumulation of:
+- the package's `leanOptions`
+- the package's `moreServerOptions`
+- the library's `leanOptions`
+- the library's `moreServerOptions`
+-/
+@[inline] def serverOptions (self : LeanLib) : Array LeanOption :=
+  self.pkg.moreServerOptions ++ self.config.moreServerOptions
```

Comment:
```suggestion
  self.pkg.moreServerOptions ++ self.config.leanOptions ++ self self.config.moreServerOptions
```

Reply:
@mhuisi Yes. They are combined for all derivative types (e.g., `Package`, `LeanLib`, `LeanExe`, etc.). After all, that is what this definition is doing, combining them for the `LeanLib` type. What makes `LeanLib` different from `Package` is that `Package` just needs too combine its two user-provided `config.*` options whereas `LeanLib` needs to combine its user-povided `config.*` options plus the combined option from `Package`.

Reply:
Ok. I somewhat disagree about the derivative types being the right place to do this (I'd probably expect it on `LeanConfig` with a name that captures the notion that it is not simply e.g. the `moreLeanArgs` field from the corresponding config), but will happily implement your suggestion :)

Reply:
@mhuisi That is another design that I had not really given much thought to before! It is worth considering for a future refactor.